src/shared/components/palette/collection/write-collection-field.component.ts

```diff
@@ -231,10 +231,44 @@ export class WriteCollectionFieldComponent extends AbstractFieldWriteComponent i
 
   private removeItem(index: number): void {
     this.collItems.splice(index, 1);
+    this.resetIds(index);
     this.caseField.value.splice(index, 1);
     this.formArray.removeAt(index);
   }
 
+  private resetIds(index: number) {
+    for (let i = index; i < this.collItems.length; i++) {
+      const counter = i  + 1;
+      if (this.collItems[i].index && this.collItems[i].index === counter) {
+        this.collItems[i].index = i;
+      }
+
+      if (this.collItems[i].caseField && this.collItems[i].caseField.id
```

Comment:
There is a huge amount going on in this if statement. It should be split out and could null coalesnce help out here reduce the multiple && checks.